os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -133,6 +133,37 @@ static int binary_manager_read_header(int bin_idx, int part_idx, binary_header_t
 	return ERROR;
 }
 
+/* Check for an update binary */
+static int binary_manager_check_update_binary(char *bin_name)
+{
+	int ret;
+	int version;
+	int bin_idx;
+	binary_header_t header_data;
+
+	if (bin_name == NULL) {
+		bmdbg("Invalid bin_name %s\n", bin_name);
+		return BINMGR_INVALID_PARAM;
+	}
+
+	bin_idx = binary_manager_get_index_with_name(bin_name);
+	if (bin_idx < 0) {
+		bmdbg("binary %s is not registered\n", bin_name);
+		return BINMGR_NOT_FOUND;
+	}
+
+	ret = binary_manager_read_header(bin_idx, BIN_USEIDX(bin_idx) ^ 1, &header_data);
+	if (ret == OK) {
+		version = (int)atoi(header_data.bin_ver);
+		/* Do not reload if it is already latest version */
+		if (version <= (int)atoi(BIN_VER(bin_idx))) {
+			bmdbg("Already latest version\n");
+			return BINMGR_ALREADY_UPDATED;
+		}
+	}
```

Comment:
If ret < 0, there is no valid binary in another partition. It means currently running binary would be reloaded by binary_manager_reload.  Is it intended for restarting binary functionality?

Reply:
Thanks, I added